core/src/main/java/hudson/model/Node.java

```diff
@@ -298,20 +298,30 @@ public OfflineCause getTemporaryOfflineCause() {
     public TagCloud<LabelAtom> getLabelCloud() {
         return new TagCloud<>(getAssignedLabels(), Label::getTiedJobCount);
     }
+
+    /**
+     * @return An immutable set of LabelAtom associated with the current node label.
+     */
+    @NonNull
```

Comment:
This could also be restricted I suppose. (In practice I do not know of any `Node` subclasses besides `Jenkins` and `Slave`.)